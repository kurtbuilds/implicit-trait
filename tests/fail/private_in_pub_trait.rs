/// Can't compile because it puts a private struct in a pub trait
use implicit_trait::implicit_trait;

pub(crate) struct Bar;

struct Foo;

#[implicit_trait(pub)]
impl FooExt for Foo {
    fn bar(&self) -> Bar {
        Bar
    }
}


fn main() {
}