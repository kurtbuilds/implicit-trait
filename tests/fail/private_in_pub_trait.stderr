error[E0446]: private type `Bar` in public interface
 --> tests/fail/private_in_pub_trait.rs:8:1
  |
4 | pub(crate) struct Bar;
  | --------------------- `Bar` declared as private
...
8 | #[implicit_trait(pub)]
  | ^^^^^^^^^^^^^^^^^^^^^^ can't leak private type
  |
  = note: this error originates in the attribute macro `implicit_trait` (in Nightly builds, run with -Z macro-backtrace for more info)
